include/seqan3/alignment/pairwise/align_pairwise.hpp

```diff
@@ -38,10 +38,92 @@
+ * incremented.
+ *
+ * ### Example
+ *
+ * The following snippets demonstrate the single element and the range based interface.
+ *
+ * \include test/snippet/alignment/pairwise/align_pairwise.cpp
+ *
+ * ### Exception
+ *
+ * Might throw std::bad_alloc if it fails to allocate the alignment matrix or seqan3::invalid_alignment_configuration
+ * if the configuration is invalid.
+ *
+ * ### Complexity
+ *
+ * The complexity depends on the configured algorithm. For the \ref seqan3::align_cfg::edit "edit distance"
+ * the following worst cases over two input sequences of size \f$ N \f$ can be assumed:
+ *
+ * | Component        | Runtime           | Space            |
+ * |------------------|-------------------|------------------|
+ * | Score only       | \f$ O(N*N/w) \f$  | \f$ O(w) \f$     |
+ * | Back coordinate  | \f$ O(N*N/w) \f$  | \f$ O(w) \f$     |
+ * | Begin coordinate | \f$ O(N*N/w) \f$  | \f$ O(N*N/w) \f$ |
+ * | Alignment        | \f$ O(N*N/w) \f$  | \f$ O(N*N/w) \f$ |
+ *
+ * \f$ w \f$ is the size of a machine word.
+ *
+ * For all other algorithms that compute the standard dynamic programming algorithm the following worst cases hold:
+ *
+ * | Component        | Runtime         | Space          |
+ * |------------------|-----------------|----------------|
+ * | Score only       | \f$ O(N^2) \f$  | \f$ O(N) \f$   |
+ * | Back coordinate  | \f$ O(N^2) \f$  | \f$ O(N) \f$   |
+ * | Begin coordinate | \f$ O(N^2) \f$  | \f$ O(N^2) \f$ |
+ * | Alignment        | \f$ O(N^2) \f$  | \f$ O(N^2) \f$ |
+ *
+ * ### Thread safety
+ *
+ * As long as no shared state is passed to the algorithm via the configuration object, the algorithm is thread safe and
+ * can be invoked in a concurrent environment.
+ */
+template <typename sequence_t,
+          typename alignment_config_t>
 //!\cond
+    requires tuple_like_concept<sequence_t> &&
+             detail::is_type_specialisation_of_v<remove_cvref_t<alignment_config_t>, configuration>
+//!\endcond
+constexpr auto align_pairwise(sequence_t && seq, alignment_config_t && config)
+{
+    static_assert(std::tuple_size_v<std::remove_reference_t<sequence_t>> == 2,
+                  "Expects exactly two sequences for pairwise alignments.");
+
+    return align_pairwise(ranges::view::single(std::forward<sequence_t>(seq)) | std::view::common,
+                          std::forward<alignment_config_t>(config));
+}
+
+//!\copydoc align_pairwise
 template <std::ranges::InputRange sequence_t, typename alignment_config_t>
+//!\cond
     requires detail::is_type_specialisation_of_v<remove_cvref_t<alignment_config_t>, configuration> &&
              tuple_like_concept<value_type_t<std::ranges::iterator_t<std::remove_reference_t<sequence_t>>>>
+//!\endcond
 constexpr auto align_pairwise(sequence_t && seq, alignment_config_t const & config)
 {
     // Wrap in persist to make the code also work with temporaries that are not views.
```

Comment:
Why are you allowing this? Won't the result range outlive the temporaries?

Reply:
Well, because if I only compute the score or the coordinates I don't need the sequences afterwards. But this needs more checking on the call site. But this will be part of other fixes and details

Reply:
now this is safe because it allows rvalue non-view ranges if the element types of the returned tuple are ViewableRanges.